docs/details/vision.dox

```diff
@@ -133,20 +133,70 @@ equal to the number of features contained in the query array.
 \defgroup cv_func_nearest_neighbour nearestNeighbour
 \ingroup featmatcher_mat
 
-\brief Nearest Neighbour
-
-Calculates nearest distances between two 2-dimensional arrays containing
-features based on the type of distance computation chosen. Currently, \ref
-AF_SAD (sum of absolute differences), \ref AF_SSD (sum of squared differences)
-and \ref AF_SHD (hamming distance) are supported.
-One of the arrays containing the training data and the other the
-query data. One of the dimensions of the both arrays must be equal among them,
-identifying the length of each feature. The other dimension indicates the
-total number of features in each of the training and query arrays. Two
-1-dimensional arrays are created as results, one containg the smallest N
-distances of the query array and another containing the indices of these
-distances in the training array. The resulting 1-dimensional arrays have length
-equal to the number of features contained in the query array.
+\brief Determine the nearest neighbouring points to a given set of points
+
+In this function, a "point" is simply a geometric point's coordinates in an
+n-dimensional space. The coordinates are described along dim0 (much like a
```

Comment:
In this function, no, because if `dist_dim` is 1, then you could look at the coordinates as a row vector

Reply:
Well, the current sentence does cause such confusion then.

Reply:
Yeah that actually bothered me too, but I thought that it's better to stick to one convention and wait until later to say "the other way could be done as well" so that I don't have to keep saying "dim0 (or dim1)" and "dim1(dim0)" repeatedly throughout the doc.

In any case I did restructure this paragraph and the one on the last in order to (hopefully) solve this dilemma.